src/ensembl/src/content/app/custom-download/components/checkbox-with-textfields/CheckboxWithTextfields.tsx

```diff
@@ -1,115 +1,148 @@
 import React, { useState, useEffect } from 'react';
 import Checkbox from 'src/shared/components/checkbox/Checkbox';
-import styles from './CheckboxWithTextfields.scss';
 
 import ImageButton from 'src/shared/components/image-button/ImageButton';
-import { ReactComponent as AddIcon } from 'static/img/browser/zoom-in.svg';
-import PasteOrUpload from '../paste-or-upload/PasteOrUpload';
+import { ReactComponent as RemoveIcon } from 'static/img/shared/clear.svg';
+
+import Textarea from 'src/shared/components/textarea/Textarea';
+import Upload, { ReadFile } from 'src/shared/components/upload/Upload';
+import { nextUuid } from 'src/shared/helpers/uuid';
+
+import styles from './CheckboxWithTextfields.scss';
 
 export type CheckboxWithTextfieldsProps = {
-  values: string[];
+  textValue: string;
+  files: ReadFile[];
   label: string;
-  allowMultiple: boolean;
   disabled?: boolean;
-  onChange: (values: string[]) => void;
+  onFilesChange: (files: ReadFile[]) => void;
+  onReset: () => void;
+  onTextChange: (value: string) => void;
 };
 
 const CheckboxWithTextfields = (props: CheckboxWithTextfieldsProps) => {
   const [isChecked, setIsChecked] = useState(false);
-  const [values, setValues] = useState<(string | null)[]>(props.values);
 
-  const [shouldShowAddButton, setShowAddButton] = useState(
-    Boolean(values[values.length - 1])
-  );
+  const [isTextareaShown, showTextarea] = useState(false);
+
+  const [files, setFiles] = useState<ReadFile[]>([]);
 
   useEffect(() => {
-    if (!props.values.length) {
-      setValues([null]);
-    } else {
-      setValues(props.values);
-    }
+    setFiles(props.files);
 
-    setIsChecked(props.values.length > 0);
-    setShowAddButton(Boolean(props.values[props.values.length - 1]));
-  }, [props.values]);
+    let checkedStatus = false;
 
-  const handleCheckboxOnChange = (isChecked: boolean) => {
-    setIsChecked(isChecked);
-    if (!isChecked && values.length > 0) {
-      props.onChange([]);
+    if (props.textValue && props.textValue.length) {
+      showTextarea(true);
+      checkedStatus = true;
     }
-  };
 
-  const addEntry = () => {
-    setShowAddButton(false);
-    setValues([...values, null]);
-  };
-
-  const handleOnChange = (value: string, index: number) => {
-    const newValues: string[] = [...values] as string[];
-    newValues[index] = value;
+    if (props.files && props.files.length) {
+      checkedStatus = true;
+    }
 
-    setShowAddButton(Boolean(newValues[newValues.length - 1]));
+    setIsChecked(checkedStatus);
```

Comment:
technically, this should only be done if checkedStatus is different from isChecked

Reply:
That's done